File: server/viewmodels/changelog.js
const UserViewModel = require('./user')

module.exports = class ChangeLogViewModel {
  constructor () {
    this.id = 0
    this.field = ''
    this.fromValue = null
    this.toValue = null
    this.author = null
    this.timestamp = new Date()
  }

  static createFromLocal (obj) {
    const log = new ChangeLogViewModel()
    if (obj) {
      log.id = obj.id
      log.author = UserViewModel.createFromLocal(obj.author)
      log.timestamp = new Date(obj.timestamp)
      log.field = obj.field
      log.fromValue = obj.oldValue
      log.toValue = obj.newValue
    }
    return log
  }

  static createFromJira (obj) {
    let result = []
    if (obj) {
      for (let item of obj.items) {
        const log = new ChangeLogViewModel()
        log.id = obj.id
        log.author = UserViewModel.createFromJira(obj.author)
        log.timestamp = new Date(obj.created)
        log.field = item.field
        log.fromValue = item.fromString
        log.toValue = item.toString
        result.push(log)
      }
    }
    return result
  }
}

File: server/model/changelog.py
from ._base import BaseModel
from .user import User


class ChangeLog(BaseModel):

    def __init__(self, id_=0, field='', oldValue='', newValue='', timestamp=''):
        super().__init__(id_)
        self.field = field
        self.oldValue = oldValue
        self.newValue = newValue
        self.timestamp = timestamp
        self.author = None

    @classmethod
    def from_db_dict(cls, obj):
        return cls(
            obj.get('id'),
            obj.get('field'),
            obj.get('oldValue'),
            obj.get('newValue'),
            obj.get('timestamp'),
        )

    def to_viewmodel(self):
        return {
            'id': self.id,
            'field': self.field,
            'fromValue': self.oldValue,
            'toValue': self.newValue,
            'timestamp': self.timestamp,
            'author': self.author.to_viewmodel(),
        }
